x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_actions.tsx

```diff
@@ -37,31 +45,44 @@ export function RuleActions({ ruleActions, actionTypeRegistry }: RuleActionsProp
         </EuiText>
       </EuiFlexItem>
     );
+  }
 
-  function getActionIconClass(actionGroupId?: string): IconType | undefined {
+  const getActionIconClass = (actionGroupId?: string): IconType | undefined => {
     const actionGroup = actionTypeRegistry.list().find((group) => group.id === actionGroupId);
     return typeof actionGroup?.iconClass === 'string'
       ? actionGroup?.iconClass
       : suspendedComponentWithProps(actionGroup?.iconClass as React.ComponentType);
-  }
+  };
+
+  const getActionName = (actionTypeId?: string) => {
+    const actionConnector = actionConnectors.find((connector) => connector.id === actionTypeId);
+    return actionConnector && actionConnector.name;
+  };
+
   if (isLoadingActionConnectors) return <EuiLoadingSpinner size="s" />;
   return (
     <EuiFlexGroup direction="column" gutterSize="none">
-      {actionConnectors.map(({ actionTypeId, name }) => (
-        <EuiFlexItem key={actionTypeId}>
-          <EuiFlexGroup alignItems="center" gutterSize="s" component="span">
-            <EuiFlexItem grow={false}>
-              <EuiIcon size="m" type={getActionIconClass(actionTypeId) ?? 'apps'} />
-            </EuiFlexItem>
-            <EuiFlexItem>
-              <EuiText data-test-subj={`actionConnectorName-${name}`} size="s">
-                {name}
-              </EuiText>
-            </EuiFlexItem>
-          </EuiFlexGroup>
-          <EuiSpacer size="s" />
-        </EuiFlexItem>
-      ))}
+      {ruleActions.map(({ actionTypeId, id }, index) => {
+        const actionName = getActionName(id);
+        return (
+          <EuiFlexItem key={index}>
```

Comment:
using index as the key here, while not ideal but we are no longer guaranteed unique ids as we might have actions with the same connector. Index is ok here since we're not re-ordering these elements